src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +60,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
     _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (!_isRunning.compareAndSet(false, true)) {
```

Comment:
AtomicBoolean has the nice method compareAndSet which volatile booleans do not have.  The equivalent code with volatile boolean is longer. 
```java
  if (!_isRunning) {
     _isRunning = true
     //stop implementation here
  }
```
So that's a few extra lines of code.